src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java

```diff
@@ -360,10 +399,27 @@ public void updateStageDeadlines(CaseData caseData) {
                 LocalDate dateReceived = caseData.getDateReceived();
                 LocalDate caseDeadline = caseData.getCaseDeadline();
                 LocalDate caseDeadlineWarning = caseData.getCaseDeadlineWarning();
-                LocalDate deadline = infoClient.getStageDeadline(stage.getStageType(), dateReceived, caseDeadline);
+
+                final StageTypeDto stageDefinition = infoClient.getAllStagesForCaseType(caseData.getType()).stream()
```

Comment:
Does this get all stages or stage types? Slight distinction but getAllStages... reads like it would get all active stages for a case type so sounds like it could return 000's of stages